File: cpp/program/Product.cpp
#include <iostream>
#include <string>

using namespace std;

class Product {
private:
    string idProduct;
    string name;
    string brand;
    double price;

public:
    Product(string idProduct, string name, string brand, double price) {
        this->idProduct = idProduct;
        this->name = name;
        this->brand = brand;
        this->price = price;
    }

    string getIdProduct() const {
        return idProduct;
    }

    string getName() const {
        return name;
    }

    string getBrand() const {
        return brand;
    }

    double getPrice() const {
        return price;
    }

    void setIdProduct(const string& id) {
        idProduct = id;
    }

    void setName(const string& newName) {
        name = newName;
    }

    void setBrand(const string& newBrand) {
        brand = newBrand;
    }

    void setPrice(double newPrice) {
        price = newPrice;
    }

    ~Product() {

    }
};

File: php/program/Product.php
<?php

class Product {
    protected $idProduct;
    protected $name;
    protected $brand;
    protected $price;

    public function __construct($idProduct = "", $name = "", $brand = "", $price = 0.0) {
        $this->idProduct = $idProduct;
        $this->name = $name;
        $this->brand = $brand;
        $this->price = $price;
    }

    public function getIdProduct() {
        return $this->idProduct;
    }

    public function getName() {
        return $this->name;
    }

    public function getBrand() {
        return $this->brand;
    }

    public function getPrice() {
        return $this->price;
    }

    public function setIdProduct($idProduct) {
        $this->idProduct = $idProduct;
    }

    public function setName($name) {
        $this->name = $name;
    }

    public function setBrand($brand) {
        $this->brand = $brand;
    }

    public function setPrice($price) {
        $this->price = $price;
    }
}

?>
